### DISCUSSION TITLE
Can this be used to generate F# code from Smithy?

### DISCUSSION THREAD

**Discussion by zetashift** [MoiraeSoftware/Myriad#156] [Ideas]
Smithy: https://awslabs.github.io/smithy/2.0/index.html
For a new project at work, I'm interested in "generating" services. But Smithy has no F# generator, so I was wondering how hard it would be to generate this from `.smithy` files. I'm not sure if I have the skills for it, but I'd like to see how far I can get. 
And generating F# services + OpenAPI docs sounds very nice to me.

**Comment by 7sharp9** [comment]
The short answer is yes, what I would do is first parse the smithy file to an ast/du then transform that to an F# ast.  This is pretty much how falanx works too which is just Google proto files to F#.

**Reply by zetashift** [reply]
Thank you! :D
